src/components/game.js

```diff
@@ -17,17 +17,25 @@ const prefix = sf`
 
 const gameView = (state, prev, send) => html`
     <div class="${prefix}">
-      ${canvasView((ctx, width, height) => render(ctx, width, height, state))}
+      ${canvasView((ctx, width, height, changed) => render(ctx, width, height, changed, state, send))}
     </div>
   `
 
-function render (ctx, width, height, state) {
+function render (ctx, width, height, changed, state, send) {
+  send('updateCanvas', { canvas: changed }, () => {})
+
+  const newXPos = state.canvas.pan.x + (state.canvas.zoom / 2 - width / 2)
+  const newYPos = state.canvas.pan.y + (state.canvas.zoom / 2 - height / 2)
+  const newGameWidth = width / state.canvas.zoom
+  const newGameHeight = height / state.canvas.zoom
+
```

Comment:
I think the zooming info should be kept inside of the canvas component. You could apply and restore the transform before and after each render call